Stratis.Bitcoin/Utilities/Stopwatch.cs

```diff
@@ -10,78 +10,47 @@ namespace Stratis.Bitcoin.Utilities
     /// {
     ///     // Time of anything executed here will be added to the used performance counter.
     /// }
     /// </code>
     /// </example>
-    public class Stopwatch
+    public class StopwatchDisposable : IDisposable
     {
+        /// <summary>Stopwatch to measure elapsed ticks of the code block.</summary>
+        private readonly System.Diagnostics.Stopwatch watch;
+
         /// <summary>
-        /// Helper class that allows the convenient <c>using</c> calls.
+        /// Action to execute when the measurement is done. 
+        /// <para>
+        /// This is usually a performance counter. 
+        /// The argument of the action is the number of elapsed ticks of the code block.
+        /// </para>
         /// </summary>
-        private class StopwatchDisposable : IDisposable
-        {
-            /// <summary>Stopwatch to measure elapsed ticks of the code block.</summary>
-            private readonly System.Diagnostics.Stopwatch watch;
+        private readonly Action<long> action;
 
-            /// <summary>
-            /// Action to execute when the measurement is done. 
-            /// <para>
-            /// This is usually a performance counter. 
-            /// The argument of the action is the number of elapsed ticks of the code block.
-            /// </para>
-            /// </summary>
-            private readonly Action<long> action;
-
-            /// <summary>
-            /// Creates a new disposable object and starts the time measurement.
-            /// </summary>
-            /// <param name="watch">Stopwatch to measure elapsed ticks of the code block.</param>
-            /// <param name="action">Action to execute when the measurement is done.</param>
-            public StopwatchDisposable(System.Diagnostics.Stopwatch watch, Action<long> action)
-            {
-                this.watch = watch;
-                this.action = action;
-                watch.Restart();
-            }
-
-            /// <summary>
-            /// Stops the time measurement and calls the action with the measured elapsed ticks.
-            /// </summary>
-            public void Dispose()
-            {
-                this.watch.Stop();
-                this.action(this.watch.Elapsed.Ticks);
-            }
-        }
-
-        /// <summary>Stopwatch to be used by <see cref="StopwatchDisposable"/> instances.</summary>
-        private readonly System.Diagnostics.Stopwatch watch;
-
-        /// <summary>Initializes a new instance of the object.</summary>
-        public Stopwatch()
+        /// <summary>
+        /// Creates a new disposable object and starts the time measurement.
+        /// </summary>
+        /// <param name="action">Action to execute when the measurement is done.</param>
+        public StopwatchDisposable(Action<long> action)
         {
+            this.action = action;
             this.watch = new System.Diagnostics.Stopwatch();
```

Comment:
Can even replace these 2 lines with `this.watch.StartNew();`

Reply:
yup